### PR TITLE
Add node version checking script. Fixes #983

### LINKED ISSUE #983: Warn coders when they have the the wrong version of node/npm
As discussed in #979.

### PR DESCRIPTION
**Pull request by turnerhayes**
TBD: all the places to run this

### CODE REVIEW

**Comment by turnerhayes** [comment]
For some reason, this didn't seem to show any link to #983

**Comment by turnerhayes** [comment]
Another question to consider; do we want to do exact version match? Or accept higher versions/different minor/patch versions? That is, if our current node version is set to `v10.13.0`, should we warn if the user is on `10.13.4`? `10.15.0`? `12.0.0`?

**Comment by turnerhayes** [comment]
Added check to pre-commit and lint. Keeping version checking to exact match.

File: check-node-version.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+const path = require('path');
+
+const targetVersion = fs.readFileSync(path.resolve(__dirname, '.nvmrc'), { encoding: 'utf8' })
+  // strip trailing and leading whitespace
+  .replace(/\s*$/, '').replace(/^\s*/, '');
+
+if (process.version !== targetVersion) {
+  const separator = '===========================================================';
+  console.warn(
+    separator + '\nUsing the wrong Node version; expected ' + targetVersion + ', got ' + process.version +
```
Comment (knod) [inline check-node-version.js:11]: @todo Maybe we make a wiki page about where to find info for updating and add a link to that in this message so people have a next step to take if they have the wrong version?

**Comment by knod** [comment]
It's weird. I'd think on `commit` you'd get two messages, since it also lints, but you only get one.

**Review by knod — APPROVED** [review]
Pulled and tested and seems to be working for me! I love it!

**Comment by turnerhayes** [comment]
I think that's because the pre-commit hook runs `eslint --fix`, rather than `npm run lint`

**Pull request merged**